spec/models/user/user_spec.rb

```diff
@@ -0,0 +1,59 @@
+RSpec.describe SynapsePayRest::User do
+  describe "get" do
+    it "makes GET request with appropriate headers, and oauth POST" do
+      ip = "127.0.0.1"
+      client = SynapsePayRest::Client.new(client_id: '1', client_secret: '2', ip_address: ip)
+
+      user_response_body = {
+        "_id" => "123",
+        refresh_token: 'ref1',
+        logins: [],
+        phone_numbers: [],
+        logal_name: [],
+        permission: 'unverified',
+        extra: {
+          note: 'foo',
+          supp_id: '123xyz',
+          is_business: false,
+          cip_tag: 1
+        },
+        documents: [] # base doc goes here
+      }.to_json
+
+      oauth_response_body = {
+        oauth_key: 'foo123',
+        expires_in: 10 # no idea what this type is
+      }.to_json
+
+      stub_request(:post, "https://uat-api.synapsefi.com/v3.1/oauth/123").
```

Comment:
Needed?

Reply:
yes, interestingly the library issues an oauth refresh every time you fetch a user, unless it is a batch fetch